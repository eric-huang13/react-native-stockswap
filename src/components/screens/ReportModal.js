import React, {Component} from 'react';
import {Text, StyleSheet, View, TouchableOpacity} from 'react-native';

export default class ReportModal extends Component {
  render() {
    const {reportModal} = this.props;
    return (
      <View style={style.reportModalContainer}>
        <TouchableOpacity onPress={() => reportModal(false)}>
          <Text style={style.closeButton}>Close</Text>
        </TouchableOpacity>
        <Text style={style.optionTextHeader}>
          Why would you like to report this post?
        </Text>
        <View style={style.innerReportContainer}>
          <Text style={style.optionText}>Option 1</Text>
          <Text style={style.optionText}>Option 2</Text>
          <Text style={style.optionText}>Option 3</Text>
        </View>
      </View>
    );
  }
}

const style = StyleSheet.create({
  reportModalContainer: {
    flex: 1,
    marginTop: 310,
    backgroundColor: '#3e4d6c',
    borderRadius: 20,
    padding: 14,
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  innerReportContainer: {
    marginTop: 30,
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '50%',
  },
  closeButton: {
    color: '#B8A0FF',
    fontFamily: 'Montserrat-Bold',
    fontSize: 14,
    marginBottom: 18,
  },
  optionTextHeader: {
    color: 'white',
    fontSize: 14,
    marginHorizontal: 12,
    paddingVertical: 2,
    fontFamily: 'Montserrat-Medium',
  },
  optionText: {
    color: 'white',
    fontSize: 12,
    marginHorizontal: 12,
    paddingVertical: 2,
    fontFamily: 'Montserrat-Medium',
  },
});
