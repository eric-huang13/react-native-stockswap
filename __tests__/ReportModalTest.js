import React from 'react';
import ReportModal from '../src/components/HomeTabComponents/ReportModal';
import ShallowRenderer from 'react-test-renderer/shallow'; // ES6


jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);


  test('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<ReportModal />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });