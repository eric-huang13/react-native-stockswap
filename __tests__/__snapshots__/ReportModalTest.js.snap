// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3e4d6c",
      "borderRadius": 20,
      "elevation": 5,
      "flex": 1,
      "marginTop": 310,
      "padding": 14,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 4,
    }
  }
>
  <ForwardRef
    onPress={[Function]}
  >
    <Text
      style={
        Object {
          "color": "#B8A0FF",
          "fontFamily": "Montserrat-Bold",
          "fontSize": 19.77372262773723,
          "marginBottom": 18,
        }
      }
    >
      Close
    </Text>
  </ForwardRef>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "Montserrat-Medium",
        "fontSize": 19.77372262773723,
        "marginHorizontal": 12,
        "paddingVertical": 2,
      }
    }
  >
    Why would you like to report this post?
  </Text>
  <View
    style={
      Object {
        "flexDirection": "column",
        "height": "50%",
        "justifyContent": "space-between",
        "marginTop": 30,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "Montserrat-Medium",
          "fontSize": 16.94890510948905,
          "marginHorizontal": 12,
          "paddingVertical": 2,
        }
      }
    >
      Option 1
    </Text>
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "Montserrat-Medium",
          "fontSize": 16.94890510948905,
          "marginHorizontal": 12,
          "paddingVertical": 2,
        }
      }
    >
      Option 2
    </Text>
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "Montserrat-Medium",
          "fontSize": 16.94890510948905,
          "marginHorizontal": 12,
          "paddingVertical": 2,
        }
      }
    >
      Option 3
    </Text>
  </View>
</View>
`;
